src/DDTrace/Encoders/Noop.php

```diff
@@ -3,9 +3,13 @@
 namespace DDTrace\Encoders;
 
 use DDTrace\Encoder;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
 
 final class Noop implements Encoder
 {
+    use LoggerAwareTrait;
```

Comment:
Do we need this trait in `Noop` class?

Reply:
The way things are currently set up, yes, because we call setLogger() on the encoder object from within the transport. Though the Noop Encoder is a bit of an odd duck anyway in terms of when you'd need it...